src/PowerShellEditorServices.Hosting/EditorServicesRunner.cs

```diff
@@ -0,0 +1,276 @@
+        {
+            bool creatingLanguageServer = _config.LanguageServiceTransport != null;
+            bool creatingDebugServer = _config.DebugServiceTransport != null;
+            bool isTempDebugSession = creatingDebugServer && !creatingLanguageServer;
+
+            // TODO: Validate config here
+
+            // Set up information required to instantiate servers
+            HostStartupInfo hostStartupInfo = CreateHostStartupInfo();
+
+            // If we just want a temp debug session, run that and do nothing else
+            if (isTempDebugSession)
+            {
+                await RunTempDebugSession(hostStartupInfo).ConfigureAwait(false);
+                return;
+            }
+
+            // We want LSP and maybe debugging
+            // To do that we:
+            //  - Create the LSP server
+            //  - Possibly kick off the debug server creation
+            //  - Start the LSP server
+            //  - Possibly start the debug server
+            //  - Wait for the LSP server to finish
+
+            // Unsubscribe the host logger here so that the integrated console is not polluted with input after the first prompt
+            _logger.Log(PsesLogLevel.Verbose, "Starting server, deregistering host logger and registering shutdown listener");
+            foreach (IDisposable loggerToUnsubscribe in _loggersToUnsubscribe)
+            {
+                loggerToUnsubscribe.Dispose();
+            }
+
+            // Write the integrated console banner
+            _config.PSHost.UI.WriteLine("\n=== PowerShell Integrated Console ===");
+
+            PsesLanguageServer languageServer = await CreateLanguageServer(hostStartupInfo).ConfigureAwait(false);
+
+            Task<PsesDebugServer> debugServerCreation = null;
+            if (creatingDebugServer)
+            {
+                debugServerCreation = CreateDebugServerWithLanguageServer(languageServer);
+            }
+
+            languageServer.StartAsync();
+
+            if (creatingDebugServer)
+            {
+                StartDebugServer(debugServerCreation);
+            }
+
+            await languageServer.WaitForShutdown().ConfigureAwait(false);
+
+            // Resubscribe host logger to log shutdown events to the console
+            _logger.Subscribe(new PSHostLogger(_config.PSHost.UI));
+        }
+
+        private async Task RunTempDebugSession(HostStartupInfo hostDetails)
+        {
+            _logger.Log(PsesLogLevel.Diagnostic, "Running temp debug session");
+            PsesDebugServer debugServer = await CreateDebugServerForTempSession(hostDetails).ConfigureAwait(false);
+            _logger.Log(PsesLogLevel.Verbose, "Debug server created");
+            await debugServer.StartAsync().ConfigureAwait(false);
+            _logger.Log(PsesLogLevel.Verbose, "Debug server started");
+            await debugServer.WaitForShutdown().ConfigureAwait(false);
+            return;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this redundant jump.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)